OpenOversight/tests/routes/test_officer_and_department.py

```diff
@@ -507,6 +507,46 @@ def test_admin_can_add_new_officer(mockdata, client, session):
         assert officer.gender == 'M'
 
 
+def test_admin_can_add_new_officer_with_unit(mockdata, client, session):
+    with current_app.test_request_context():
+        login_admin(client)
+        department = random.choice(dept_choices())
+        unit = random.choice(unit_choices())
+        links = [
+            LinkForm(url='http://www.pleasework.com', link_type='link').data,
+            LinkForm(url='http://www.avideo/?v=2345jk', link_type='video').data
+        ]
+        job = Job.query.filter_by(department_id=department.id).first()
+        form = AddOfficerForm(first_name='Test',
+                              last_name='McTesterson',
+                              middle_initial='T',
+                              race='WHITE',
+                              gender='M',
+                              star_no=666,
+                              job_title=job.id,
+                              unit=unit.id,
+                              department=department.id,
+                              birth_year=1990,
+                              links=links)
+
+        data = process_form_data(form.data)
+
+        rv = client.post(
+            url_for('main.add_officer'),
+            data=data,
+            follow_redirects=True
+        )
+
+        assert 'New Officer McTesterson added' in rv.data.decode('utf-8')
+
+        # Check the officer was added to the database
+        officer = Officer.query.filter_by(
+            last_name='McTesterson').one()
+        assert officer.first_name == 'Test'
+        assert officer.race == 'WHITE'
+        assert officer.gender == 'M'
+
```

Comment:
Can you add an `assert` here and on the other tests that make sure that the unit was stored in the db too. `unit` is not a field stored on `officer` directly, so maybe something of the form
```
assert Assignment.query.filter_by(baseofficer=officer, unit=unit).one()
```